pyiceberg/view/metadata.py

```diff
@@ -0,0 +1,89 @@
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from typing import Dict, List, Literal, Optional
+
+from pydantic import Field, field_validator
+
+from pyiceberg.schema import Schema
+from pyiceberg.typedef import IcebergBaseModel, Properties
+from pyiceberg.types import transform_dict_value_to_str
+
+
+class ViewRepresentation(IcebergBaseModel):
+    """Represents the SQL query that defines the view."""
+
+    type: Literal["sql"] = Field()
+    """A string that indicates the type of representation. Must be `sql`"""
+    sql: str = Field()
+    """A string that contains the SQL text of the view definition."""
+    dialect: str = Field()
+    """The dialect of the SQL, e.g. `spark`, `trino`, `presto`."""
+
+
+class ViewVersion(IcebergBaseModel):
+    """A version of the view definition."""
+
+    version_id: int = Field(alias="version-id")
+    """ID for the version"""
+    schema_id: int = Field(alias="schema-id")
+    """ID of the schema for the view version"""
+    timestamp_ms: int = Field(alias="timestamp-ms")
+    """Timestamp when the version was created (ms from epoch)"""
+    summary: Dict[str, str] = Field()
+    """A string to string map of summary metadata about the version"""
+    representations: List[ViewRepresentation] = Field()
+    """A list of representations for the view definition"""
+    default_catalog: Optional[str] = Field(alias="default-catalog", default=None)
+    """Catalog name to use when a reference in the SELECT does not contain a catalog"""
+    default_namespace: List[str] = Field(alias="default-namespace")
+    """Namespace to use when a reference in the SELECT is a single identifier"""
+
+
+class ViewVersionLogEntry(IcebergBaseModel):
+    """A log entry of a view version change."""
+
+    timestamp_ms: int = Field(alias="timestamp-ms")
+    """Timestamp when the version was created (ms from epoch)"""
+    version_id: int = Field(alias="version-id")
+    """ID for the version"""
+
+
+class ViewMetadata(IcebergBaseModel):
+    """The metadata for a view."""
+
+    view_uuid: str = Field(alias="view-uuid")
+    """A UUID that identifies the view, generated when the view is created."""
+    format_version: Literal[1] = Field(alias="format-version")
+    """An integer version number for the view format; must be 1"""
+    location: str = Field()
```

Comment:
Is it possible to raise a friendlier error message?
https://github.com/apache/iceberg/blob/83cc7d8749f2ba7ed7f0bff09e7bcf10d77785e8/core/src/main/java/org/apache/iceberg/view/ViewMetadata.java#L130

```py
from pydantic import BaseModel, Field
from typing import Literal

class MyModel(BaseModel):
    my_literal: Literal[1] = Field(alias="my-literal")

MyModel(my_literal=1)
```
Will produce
```
pydantic_core._pydantic_core.ValidationError: 1 validation error for MyModel
my-literal
  Field required [type=missing, input_value={'my_literal': 1}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.11/v/missing
```